pkg/kubelet/kubelet.go

```diff
@@ -1155,12 +1155,22 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
 }
 
 // getPullSecretsForPod inspects the Pod and retrieves the referenced pull secrets
-// TODO transitively search through the referenced service account to find the required secrets
 // TODO duplicate secrets are being retrieved multiple times and there is no cache.  Creating and using a secret manager interface will make this easier to address.
 func (kl *Kubelet) getPullSecretsForPod(pod *api.Pod) ([]api.Secret, error) {
-	pullSecrets := []api.Secret{}
+	secretRefs := []api.LocalObjectReference{}
+	secretRefs = append(secretRefs, pod.Spec.ImagePullSecrets...)
+
+	if len(pod.Spec.ServiceAccount) > 0 {
+		serviceAccount, err := kl.kubeClient.ServiceAccounts(pod.Namespace).Get(pod.Spec.ServiceAccount)
+		if err != nil {
```

Comment:
if we've decided that we're merging, should a failed service account lookup fail or continue? worst case, the pull fails because we didn't pick up the service account's pull secrets, right?